storage/http_client.go

```diff
@@ -593,6 +593,31 @@ func (c *httpStorageClient) RestoreObject(ctx context.Context, params *restoreOb
 	return newObject(obj), err
 }
 
+func (c *httpStorageClient) MoveObject(ctx context.Context, params *moveObjectParams, opts ...storageOption) (*ObjectAttrs, error) {
```

Comment:
What happen if we call this same API for flat bucket, I mean what error it will return?

Reply:
It's not the best error -- I am filing a bug with an internal team.

GCSFuse should avoid using MoveObject unless you know it's an HNS bucket.